src/components/MAPI/clusters/ClusterListItemDetails.tsx

```diff
@@ -0,0 +1,63 @@
+import { Box, Text } from 'grommet';
+import { useHttpClient } from 'lib/hooks/useHttpClient';
+import { GenericResponse } from 'model/clients/GenericResponse';
+import * as capiv1alpha3 from 'model/services/mapi/capiv1alpha3';
+import * as capzv1alpha3 from 'model/services/mapi/capzv1alpha3';
+import * as metav1 from 'model/services/mapi/metav1';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import { getLoggedInUser } from 'stores/main/selectors';
+import useSWR from 'swr';
+
+interface IClusterListItemDetailsProps {
+  cluster: capiv1alpha3.ICluster;
+}
+
+const ClusterListItemDetails: React.FC<IClusterListItemDetailsProps> = ({
+  cluster,
+}) => {
+  const client = useHttpClient();
+  const user = useSelector(getLoggedInUser);
+
+  // Use type intersection for multiple cluster types.
+  const { data, error } = useSWR<capzv1alpha3.IAzureCluster, GenericResponse>(
+    capiv1alpha3.getClusterInfraRefKey(user, cluster),
+    capiv1alpha3.getClusterInfraRef(client, user!, cluster)
+  );
+
+  if (
+    metav1.isStatusError(error?.data, metav1.K8sStatusErrorReasons.NotFound)
```

Comment:
How to check if an error is of a specific K8s API error type. In this block the error type will now get the reason and error status code.

![image](https://user-images.githubusercontent.com/13508038/111289035-08985b00-8645-11eb-9902-8d242f58b8d6.png)